components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3894,6 +3894,12 @@ void RewardsServiceImpl::ClearAllNotifications() {
 
 void RewardsServiceImpl::CompleteReset(SuccessCallback callback) {
   notification_service_->DeleteAllNotifications(true);
+
+  // Close any open files before deleting them below (required on
+  // Windows)
+  file_task_runner_->DeleteSoon(FROM_HERE, rewards_database_.release());
```

Comment:
Closing the database wasn't a great option, as it led to some cascading problems in other parts of the code that weren't expecting or testing for a closed database. Releasing the database seems safer and we handle it properly elsewhere (and it's already being done later in the reset process).

My understanding of `DeleteSoon` is that it fires the release immediately after unwinding the current function stack and will therefore always run before the task we post to delete the files. Let me know if that doesn't sound like a good approach though.